workflow/controller/controller.go

```diff
@@ -636,6 +610,11 @@ func (wfc *WorkflowController) addWorkflowInformerHandler() {
 				}
 			},
 			UpdateFunc: func(old, new interface{}) {
+				oldWf, newWf := old.(*unstructured.Unstructured), new.(*unstructured.Unstructured)
+				if oldWf.GetResourceVersion() == newWf.GetResourceVersion() {
+					wfc.metrics.WorkflowResourceVersionRepeated()
```

Comment:
keep this check, but you should return here

Reply:
Will do. Wasn't sure we wanted that functionality yet